instrumentedProfilingLibrary/Exercises/time_profiler_constructor_destructor.cpp

```diff
@@ -0,0 +1,68 @@
+#include<iostream>
+#include<chrono>
+#include<cmath>
+
+
+using namespace std;
+
+class time_profiler{
```

Comment:
BTW, move the time_profiler class declaration to a `.h` separate header and keep the implementation in a separate `.c` file, eg. `time_profiler.hh` and `time_profiler.cpp`. In the main file, just include the `time_profiler` header.
Also, declare the `time_profiler` class in the `igtk` namespace :). That way, you'll learn how to juggle with namespace bettter.

Reply:
The first part has been attempted, I separated the initial code into parts comprising of  `.h` and `.cpp` for the `time_profiler` and `operations`. 
I am still working the second part  of declare the `time_profiler` class in the `igtk` namespace

Reply:
brushing up the use of namespaces other than the one am used to of `std`
I came up with this as an example:
```
namespace igtk{
	class time_profiler;
}
class igtk::time_profiler{
	public:
		time_profiler(std::string blockName);
	private:
		std::string codeBlock;
		std::chrono::time_point<std::chrono::high_resolution_clock> codeBlockStartTime;
};
igtk::time_profiler::time_profiler(string blockName): codeBlock(move(blockName)),codeBlockStartTime(chrono::high_resolution_clock::now()){

}
```

Reply:
Although C++ allows you to prefix with a namespace scope a class definition, it's more common to simply define it inside a enclosing namespace, i.e.
```
namespace igtk
{
 class time_profiler
...
}
```